### PR TITLE
Non-square block search levels

### PR DESCRIPTION
**Pull request by NaderMahdi**
**Added support non-square block (nsq) search levels for better speed-quality trade-offs**

```
// NSQ search Level                               Settings
// 0                                                         OFF
// 1                                                         Allow only NSQ Intra-FULL if parent SQ is intra-coded and vice versa.
// 2                                                         Allow only NSQ Inter-NEAREST/NEAR/GLOBAL if parent SQ has no coeff
// 3                                                         Allow only NSQ Intra-FULL and Inter-NEWMV if parent SQ is NEWMV
// 4                                                         Allow only NSQ Inter-FULL and Intra-Z3 if parent SQ is intra-coded
// 5                                                         Allow NSQ Intra-FULL and Inter-FULL
```

### CODE REVIEW

File: Source/Lib/Codec/EbPictureDecisionProcess.c [hunk 1]
```diff
@@ -658,6 +658,26 @@ EbErrorType signal_derivation_multi_processes_oq(
...
+    // NSQ search Level                               Settings
+    // 0                                              OFF
+    // 1                                              Allow only NSQ Intra-FULL if parent SQ is intra-coded and vice versa.
+    // 2                                              Allow only NSQ Inter-NEAREST/NEAR/GLOBAL if parent SQ has no coeff
+    // 3                                              Allow only NSQ Intra-FULL and Inter-NEWMV if parent SQ is NEWMV
+    // 4                                              Allow only NSQ Inter-FULL and Intra-Z3 if parent SQ is intra-coded
+    // 5                                              Allow NSQ Intra-FULL and Inter-FULL
+
+    picture_control_set_ptr->nsq_search_level = NSQ_SEARCH_FULL;
+
```
Comment (mafonsoN) [inline Source/Lib/Codec/EbPictureDecisionProcess.c]: Extra space

File: Source/Lib/Codec/EbRestorationPick.c [hunk 2]
```diff
@@ -725,6 +725,31 @@ static int32_t count_sgrproj_bits(SgrprojInfo *sgrproj_info,
...
+    case 3:
+        step = 4;
+        break;
+    case 4:
+        step = 16;
+        break;
+    default:
+        step = 16;
+        break;
+
```
Comment (mafonsoN) [inline Source/Lib/Codec/EbRestorationPick.c]: Extra space

File: Source/Lib/Codec/EbPictureDecisionProcess.c (hunk 1 above)
Comment (NaderMahdi) [reply Source/Lib/Codec/EbPictureDecisionProcess.c]: Done

File: Source/Lib/Codec/EbRestorationPick.c (hunk 2 above)
Comment (NaderMahdi) [reply Source/Lib/Codec/EbRestorationPick.c]: Done

**Pull request merged**